contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py

```diff
@@ -128,95 +128,146 @@ def _transcribe_property(prop):
+
+    mob_chain = [source_clip.mob]
+    source_clip_chain = []
+    timecode_chain = []
+    source_start_multiplier = None
+    slot_id = source_clip.slot_id
+
+    for mob in mob_chain:
+        if hasattr(mob, 'descriptor') \
+                and mob.descriptor.name in ['TapeDescriptor', 'ImportDescriptor']:
+            timecode_chain.append(_find_mobslot_timecode(mob, 1))
+            # Doing this so mob_chain and source_clip_chain will always have the same
+            # number of values for zip function used below
+            source_clip_chain.append(None)
+            continue
+        elif not mob:
             continue
 
-    return mobs
-
-
-def timecode_values_are_same(timecodes):
-    """
-    A SourceClip can have multiple timecode objects (for example an auxTC24
-    value that got added via the Avid Bin column). As long as they have the
-    same start and length values, they can be treated as being the same.
-    """
-    if len(timecodes) == 1:
-        return True
-
-    start_set = set()
-    length_set = set()
-
-    for timecode in timecodes:
-        start_set.add(timecode.getvalue('Start'))
-        length_set.add(timecode.getvalue('Length'))
-
-    # If all timecode objects are having same start and length we can consider
-    # them equivalent.
-    if len(start_set) == 1 and len(length_set) == 1:
-        return True
-
-    return False
-
-
-def _extract_timecode_info(mob):
-    """Given a mob with a single timecode slot, return the timecode and length
-    in that slot as a tuple
-    """
-    timecodes = [slot.segment for slot in mob.slots
-                 if isinstance(slot.segment, aaf2.components.Timecode)]
-
-    # Only use timecode if we have just one or multiple ones with same
-    # start/length.
-    if timecode_values_are_same(timecodes):
-        timecode = timecodes[0]
-        timecode_start = timecode.getvalue('Start')
-        timecode_length = timecode.getvalue('Length')
-
-        if timecode_start is None or timecode_length is None:
-            raise otio.exceptions.NotSupportedError(
-                "Unexpected timecode value(s) in mob named: `{}`."
-                " `Start`: {}, `Length`: {}".format(mob.name,
-                                                    timecode_start,
-                                                    timecode_length)
-            )
-
-        return timecode_start, timecode_length
-    elif len(timecodes) > 1:
-        raise otio.exceptions.NotSupportedError(
-            "Error: mob has more than one timecode slot with different values."
-            " This is currently not supported by the AAF adapter. Found:"
-            " {} slots,  mob name is: '{}'".format(len(timecodes), mob.name)
-        )
-    else:
-        return None
+        slot = [s for s in mob.slots if s.slot_id == slot_id]
+        slot = slot[0] if len(slot) > 0 else None
+        result = _find_source_clip(slot.segment) if slot else (None, None)
```

Comment:
From Zoom code review: this is where the code attempts to drill into the SourceClip to find the details of what's inside it